homeassistant/components/modbus/binary_sensor.py

```diff
@@ -16,52 +16,72 @@
 
 _LOGGER = logging.getLogger(__name__)
 
-CONF_COIL = "coil"
-CONF_COILS = "coils"
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_COILS): [
-            {
-                vol.Required(CONF_COIL): cv.positive_int,
-                vol.Required(CONF_NAME): cv.string,
-                vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA,
-                vol.Optional(CONF_HUB, default=DEFAULT_HUB): cv.string,
-                vol.Optional(CONF_SLAVE): cv.positive_int,
-            }
-        ]
-    }
+CONF_DEPRECATED_COIL = "coil"
+CONF_DEPRECATED_COILS = "coils"
+
+CONF_INPUTS = "inputs"
+CONF_INPUT_TYPE = "input_type"
+CONF_ADDRESS = "address"
+
+INPUT_TYPE_COIL = "coil"
+INPUT_TYPE_DISCRETE = "discrete_input"
+
+PLATFORM_SCHEMA = vol.All(
+    cv.deprecated(CONF_DEPRECATED_COILS, CONF_INPUTS),
+    PLATFORM_SCHEMA.extend(
+        {
+            vol.Required(CONF_INPUTS): [
+                vol.All(
+                    cv.deprecated(CONF_DEPRECATED_COIL, CONF_ADDRESS),
+                    vol.Schema(
```

Comment:
This schema should still accept `CONF_DEPRECATED_COIL` or else it won't pass ? 

I also don't see backwards compatability code to use CONF_DEPRECATED_COIL in the code below. Right now it warns but is not backwards compatible?

Reply:
It is backwards compatible. I'm using `deprecated()` helper function, which actually removes the deprecated key and puts a replacement key to its place. I actually checked the code how this function works. So, the schema actually validates replacement key and the value originally assigned to the deprecated key.

A nice benefit is that I don't need to change existing code (besides scheme validation). Also, in the future, removing deprecated commands should be easy to do.

To test that it really works, I tried different configurations with and without deprecated options. UI correctly displayed the sensor values each time.

Reply:
Actually, the relevant code to replace the key is in the `validator()` function which is inside the `deprecated()` function. Hopefully, this will clarify things better.

Reply:
I need to read up on the deprecated key 🤦‍♂ Thanks for clarifying.